src/main/java/sorald/processor/SoraldAbstractProcessor.java

```diff
@@ -31,49 +22,8 @@
 
     public SoraldAbstractProcessor() {}
 
-    public SoraldAbstractProcessor initResource(String originalFilesPath, File baseDir) {
-        JavaFileScanner sonarCheck = Checks.getCheckInstance(getRuleKey());
-        try {
-            List<String> filesToScan = new ArrayList<>();
-            File file = new File(originalFilesPath);
-            if (file.isFile()) {
-                filesToScan.add(file.getAbsolutePath());
-            } else {
-                try {
-                    filesToScan =
-                            FileUtils.findFilesByExtension(file, Constants.JAVA_EXT).stream()
-                                    .map(File::getAbsolutePath)
-                                    .collect(Collectors.toList());
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-            ruleViolations = RuleVerifier.analyze(filesToScan, baseDir, sonarCheck);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return this;
-    }
-
-    public SoraldAbstractProcessor initResource(List<Node> segment, File baseDir) {
-        JavaFileScanner sonarCheck = Checks.getCheckInstance(getRuleKey());
-        List<String> filesToScan = new ArrayList<>();
-        for (Node node : segment) {
-            if (node.isFileNode()) {
-                filesToScan.addAll(node.getJavaFiles());
-            } else {
-                try (Stream<Path> walk = Files.walk(Paths.get(node.getRootPath()))) {
-                    filesToScan.addAll(
-                            walk.map(x -> x.toFile().getAbsolutePath())
-                                    .filter(f -> f.endsWith(Constants.JAVA_EXT))
-                                    .collect(Collectors.toList()));
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-        }
```

Comment:
This method is not actually necessary. It's more efficient to scan the entire project for rule violations before splitting it into segments, than to scan the segments one-by-one. It doesn't matter that when repairing segment A there are also violations for segment B in the violations set of the processor, as they won't match anyway.

Reply:
This code was also duplicate, it's nice that it's gone!